sys/net/network_layer/fib/fib.c

```diff
@@ -132,7 +132,8 @@ static int fib_find_entry(fib_table_t *table, uint8_t *dst, size_t dst_size,
 
             int ret_comp = universal_address_compare(table->data.entries[i].global, dst, &match_size);
             /* If we found an exact match */
-            if ((ret_comp == 0) || (is_all_zeros_addr && (match_size == 0) && (ret_comp == 2))) {
+            if ((ret_comp == UNIVERSAL_ADDRESS_EQUAL) 
+                || (is_all_zeros_addr && (match_size == 0) && (ret_comp == UNIVERSAL_ADDRESS_IS_ALL_ZERO_ADDRESS))) {
```

Comment:
I am wondering: is it even necessary to check for `is_all_zeros_addr && (match_size == 0)` if `ret_comp == UNIVERSAL_ADDRESS_IS_ALL_ZERO_ADDRESS`?